lib/Sema/TypeChecker.h

```diff
@@ -475,7 +525,15 @@ RequirementCheckResult checkGenericArguments(
     ArrayRef<Requirement> requirements, TypeSubstitutionFn substitutions,
     SubstOptions options = None);
 
-/// A lower-level version of the above without diagnostic emission.
+/// Return the result of checking the given generic parameter substitutions
+/// against the given requirements.
+CheckGenericArgumentsResult
+checkGenericArgumentsInformative(ModuleDecl *module,
```

Comment:
Then how about `checkGenericArgumentsAndDiagnose`, or `checkGenericArgumentsForDiagnostics`? I think these versions make for a slightly better hint that the function returns data for diagnostics rather than performing them in place.

Reply:
I went with `checkGenericArgumentsAndDiagnose`.